libs/components/tabs/src/lib/modules/tabs/tabset-nav-button-disabled.pipe.ts

```diff
@@ -0,0 +1,23 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+import { SkyTabComponent } from './tab.component';
+import { SkyTabsetNavButtonType } from './tabset-nav-button-type';
+
+@Pipe({
+  name: 'skyTabsetNavButtonDisabled',
+})
+export class SkyTabsetNavButtonDisabledPipe implements PipeTransform {
+  public transform(
+    disabled: boolean | undefined,
+    buttonType?: SkyTabsetNavButtonType | string,
+    tabToSelect?: SkyTabComponent
+  ): boolean {
+    if (disabled !== undefined) {
+      return disabled;
+    } else if (buttonType === 'finish') {
+      return false;
+    }
+
+    return tabToSelect === undefined || !!tabToSelect.disabled;
```

Comment:
Checking a property value in a pure pipe could be problematic if the property value can change. Pure pipes are only called when their inputs change, so in the case where `tabToSelect.disabled` changes values but `tabToSelect` still points to the same object when the pipe was last called, this pipe won't be called. Instead of taking in `tabToSelect`, it should probably take in two `boolean` values, `tabToSelectExists` and `tabToSelectIsDisabled`. Then the template would look like this:

```
[disabled]="disabled | skyTabsetNavButtonDisabled: buttonType:!!tabToSelect:tabToSelect?.disabled"
```